crates/solidity/testing/snapshots/bindings_output/contracts/call_public_getter/input.sol

```diff
@@ -0,0 +1,9 @@
+contract Base {
+    struct Value { int x; }
+    Value public value;
+}
+contract Test {
+    function test(Base base) public {
+        base.value().x;
```

Comment:
solc produces an error for this code. Not sure I'm following?

```
TypeError: Member "x" not found or not visible after argument-dependent lookup in int256.
```

Reply:
I introduced a fix for the simplest case: a struct that has a single field. In that case, the getter would return the value of such field. If the struct has more fields, Solidity will generate a getter that returns a tuple. That case is not relevant for name binding, since you cannot chain calls to tuple results (since tuples are not a real type).

There is an additional caveat and that is: if the only field in the struct is not an elementary type, then it cannot be returned in a getter. We don't handle that case and blindly bind to the type of the first field.

For more complex cases such as the one presented in Solidity documentation, I don't think we can do anything and will need to be handled in later passes.

For reference, the complex example the present is:
```solidity
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.4.0 <0.9.0;

contract Complex {
    struct Data {
        uint a;
        bytes3 b;
        mapping(uint => uint) map;
        uint[3] c;
        uint[] d;
        bytes e;
    }
    mapping(uint => mapping(bool => Data[])) public data;
}
```
Which generates this getter:
```solidity
function data(uint arg1, bool arg2, uint arg3)
    public
    returns (uint a, bytes3 b, bytes memory e)
{
    a = data[arg1][arg2][arg3].a;
    b = data[arg1][arg2][arg3].b;
    e = data[arg1][arg2][arg3].e;
}
```

Reply:
> For more complex cases such as the one presented in Solidity documentation, I don't think we can do anything and will need to be handled in later passes.

In that case, I suggest adding it as a GitHub issue, or maybe in the "limitations" list so that we don't forget about it.

Reply:
Yes, we still bind the getter. I'll definitely add this to the limitations document.